### PR TITLE
Replace Guava multimap with hashmap

### PR DESCRIPTION
**Pull request by kunsingh**
https://github.com/canoo/dolphin-platform/issues/772

---
This change is [Reviewable](https://reviewable.io/reviews/canoo/dolphin-platform/793)

### CODE REVIEW

File: platform/dolphin-platform-remoting-common/src/main/java/com/canoo/dp/impl/remoting/BeanRepositoryImpl.java [hunk 1]
```diff
@@ -70,47 +67,43 @@ public void onEvent(PresentationModel model) {
...
         DolphinUtils.assertIsDolphinBean(beanClass);
-        beanAddedListenerMap.put(beanClass, listener);
-        return new Subscription() {
-            @Override
-            public void unsubscribe() {
-                beanAddedListenerMap.remove(beanClass, listener);
+        if (beanAddedListenerMap.containsKey(beanClass)) {
+            beanAddedListenerMap.get(beanClass).add(listener);
+        } else {
+            beanAddedListenerMap.put(beanClass, Arrays.asList(listener));
```
Comment (hendrikebbers) [inline platform/dolphin-platform-remoting-common/src/main/java/com/canoo/dp/impl/remoting/BeanRepositoryImpl.java]: Does Arrays.asList() provide a mutable list?

File: platform/dolphin-platform-remoting-common/src/main/java/com/canoo/dp/impl/remoting/BeanRepositoryImpl.java [hunk 2]
```diff
@@ -30,38 +28,37 @@
...
+                if(beanRemovedListenerMap.containsKey(bean.getClass())){
                     for (final BeanRemovedListener beanRemovedListener : beanRemovedListenerMap.get(bean.getClass())) {
                         beanRemovedListener.beanDestructed(bean);
                     }
-                    for (final BeanRemovedListener beanRemovedListener : anyBeanRemovedListeners) {
-                        beanRemovedListener.beanDestructed(bean);
-                    }
+                }
+                for (final BeanRemovedListener beanRemovedListener : anyBeanRemovedListeners) {
+                    beanRemovedListener.beanDestructed(bean);
```
Comment (hendrikebbers) [inline platform/dolphin-platform-remoting-common/src/main/java/com/canoo/dp/impl/remoting/BeanRepositoryImpl.java]: Is it right that this moved out off the inner if statement?

File: platform/dolphin-platform-remoting-common/src/main/java/com/canoo/dp/impl/remoting/BeanRepositoryImpl.java [hunk 3]
```diff
@@ -70,47 +67,43 @@ public void onEvent(PresentationModel model) {
...
 
     @Override
     public <T> Subscription addOnRemovedListener(final Class<T> beanClass, final BeanRemovedListener<? super T> listener) {
         DolphinUtils.assertIsDolphinBean(beanClass);
-        beanRemovedListenerMap.put(beanClass, listener);
-        return new Subscription() {
-            @Override
-            public void unsubscribe() {
-                beanRemovedListenerMap.remove(beanClass, listener);
+        if (beanRemovedListenerMap.containsKey(beanClass)) {
```
Comment (hendrikebbers) [inline platform/dolphin-platform-remoting-common/src/main/java/com/canoo/dp/impl/remoting/BeanRepositoryImpl.java]: Map in Java8 provides nice new methods for this pattern ;)

File: platform/dolphin-platform-remoting-common/src/main/java/com/canoo/dp/impl/remoting/BeanRepositoryImpl.java (hunk 2 above)
Comment (kunsingh) [reply platform/dolphin-platform-remoting-common/src/main/java/com/canoo/dp/impl/remoting/BeanRepositoryImpl.java]: Yes....below for is working on different DataStructure and that is a List.

File: platform/dolphin-platform-remoting-common/src/main/java/com/canoo/dp/impl/remoting/BeanRepositoryImpl.java (hunk 1 above)
Comment (kunsingh) [reply platform/dolphin-platform-remoting-common/src/main/java/com/canoo/dp/impl/remoting/BeanRepositoryImpl.java]: Yes...it return new ArrayList(value)

File: platform/dolphin-platform-remoting-common/src/main/java/com/canoo/dp/impl/remoting/BeanRepositoryImpl.java (hunk 3 above)
Comment (kunsingh) [reply platform/dolphin-platform-remoting-common/src/main/java/com/canoo/dp/impl/remoting/BeanRepositoryImpl.java]:
```
if (beanRemovedListenerMap.containsKey(beanClass)) {
    beanRemovedListenerMap.get(beanClass).add(listener);
} else {
    beanRemovedListenerMap.put(beanClass, Arrays.asList(listener));
}
```
We can convert above if as -
```
beanRemovedListenerMap.computeIfAbsent(beanClass, s -> addListener(s, listener));
```
What about else part?

Comment (hendrikebbers) [reply platform/dolphin-platform-remoting-common/src/main/java/com/canoo/dp/impl/remoting/BeanRepositoryImpl.java]: `beanRemovedListenerMap.computeIfAbsent(beanClass, s -> new ArrayList()).add(listener);`

**Pull request merged**